pkg/apis/softwarecomposition/validation/valdiation_test.go

```diff
@@ -0,0 +1,115 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/kubescape/storage/pkg/apis/softwarecomposition"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateNetworkNeighbors(t *testing.T) {
+	port80 := int32(80)
+	port1000000 := int32(1000000)
+	tests := []struct {
+		name             string
+		port             int32
+		networkNeighbors softwarecomposition.NetworkNeighbors
+		expectedErrors   field.ErrorList
+	}{
+		{
+			name: "valid",
+			networkNeighbors: softwarecomposition.NetworkNeighbors{
+				Spec: softwarecomposition.NetworkNeighborsSpec{
+					Ingress: []softwarecomposition.NetworkNeighbor{
+						{
+							Identifier: "test",
+							Ports: []softwarecomposition.NetworkPort{
+								{
+									Port:     &port80,
+									Name:     "UDP-80",
+									Protocol: "UDP",
+								},
```

Comment:
suggestion(non-blocking): use the pointer utils in K8s

K8s infrastructure provides utility functions for working with different kinds of pointers. We have the package as a dependency anyway, so we should use it.

```suggestion
	"k8s.io/apimachinery/pkg/util/validation/field"
	"k8s.io/utils/pointer"
)

func TestValidateNetworkNeighbors(t *testing.T) {
	tests := []struct {
		name             string
		port             int32
		networkNeighbors softwarecomposition.NetworkNeighbors
		expectedErrors   field.ErrorList
	}{
		{
			name: "valid",
			networkNeighbors: softwarecomposition.NetworkNeighbors{
				Spec: softwarecomposition.NetworkNeighborsSpec{
					Ingress: []softwarecomposition.NetworkNeighbor{
						{
							Identifier: "test",
							Ports: []softwarecomposition.NetworkPort{
								{
									Port:     pointer.Int32(80),
									Name:     "UDP-80",
									Protocol: "UDP",
								},
```